### PR TITLE
Support GHC 9.4

### PR DESCRIPTION
**Pull request by j6carey**
Also support various dependencies and upgrade nixpkgs.

### CODE REVIEW

**Review by Friede80 — APPROVED** [review]
Thanks for doing this work!

File: grpc-mqtt.cabal [hunk 1]
```diff
@@ -42,33 +42,33 @@ common common
...
     , net-mqtt             >= 0.8.2    && < 0.9
-    , proto3-suite         >= 0.5.2    && < 0.7
+    , proto3-suite         >= 0.5.2    && < 0.8
     , proto3-wire          >= 1.2.2    && < 1.5
-    , relude               >= 0.7.0    && < 0.8
+    , relude               >= 0.7.0    && < 1.3
     , stm                  >= 2.5.0    && < 2.6
-    , text                 >= 0.2      && < 1.3
-    , time                 >= 1.9.3    && < 1.10
+    , text                 >= 0.2      && < 1.3 || == 2.*
```
[the remark this answers is not in the document]
Comment (j6carey) [reply grpc-mqtt.cabal:64]: I've seen other packages that don't care about the v1/v2 distinction do this; I was copying their style.  But sure, it does seem unlikely anyone will make a text-1.4 with breaking changes.

[the remark this answers is not in the document]
Comment (j6carey) [reply grpc-mqtt.cabal]: Changed in e5b83dddce035211ec313efeadbf232e2ad63c59 .

File: nix/overlays/haskell.nix [hunk 2]
```diff
@@ -59,8 +73,12 @@ final: prev: {
...
           hsPkgs.hp2pretty
           hsPkgs.proto3-suite
-          hsPkgs.threadscope
-        ];
+        ] ++ (
+          # threadscope does not yet support template-haskell-2.19, the version for ghc 9.4
+          if builtins.compareVersions hsPkgs.ghc.version "9.4" < 0
+            then [hsPkgs.threadscope]
+            else []
+        );
```
[the remark this answers is not in the document]
Comment (j6carey) [reply nix/overlays/haskell.nix]: Sure; removed by e5b83dddce035211ec313efeadbf232e2ad63c59 .

**Comment by j6carey** [comment]
@riz0id , @Friede80 , I'm fine with this being merged.  But for me the "Squash and merge" button is greyed out for some reason.

**Comment by ixmatus** [comment]
@j6carey it's because we have a merge requirement on the repo that all commits be signed. I don't think that needs to be so hard and fast so I'll go ahead and merge it for you.

**Comment by j6carey** [comment]
@ixmatus , thanks.  Let me know if I should be signing commits in future.

**Pull request merged**